midterm-2023-11-06/code/5100-nenad-saldziev/script.ts

```diff
@@ -0,0 +1,188 @@
+interface albums{ 
+        name: string; // the name of the album
+        year: number; // the year the album was released
+} // an array of album objects
+
+interface metalBand{  // Author == metalBand
+    id: number;
+    name: string;
+    genre: string;
+    formed: number; // the year the band was formed
+    location : string; // the city and country where the band was formed
+    members: [ string ]; // an array of strings, each string is the name of a band member
+    album: albums[];
+}
+// Array of band[objects]
+
+type BandSorter = (first: metalBand, second: metalBand) => number;
+
+document.addEventListener("DOMContentLoaded", backsitcode)
+
+let bands: metalBand[] = []; // all authors == bands
+
+async function backsitcode() {
+    const data = await waitData();
```

Comment:
The loading of the data does not take into account the existence of the `metalBands` property in the data. +5